mesonbuild/compilers/c.py

```diff
@@ -710,3 +710,66 @@ def get_include_args(self, path: str, is_system: bool) -> T.List[str]:
         if path == '':
             path = '.'
         return ['--include_path=' + path]
+
+class TCCCCompiler(CCompiler):
```

Comment:
It looks like tcc has some support for -MD and -MF options; if I'm not mistaken these can be set at get_dependency_gen_args.

Reply:
I am looking into this, because that is causing some unit tests to fail, however it is not working as I would expect yet.

Reply:
Pushed a version with the -MD, -MF, however it causes other fails because of `assertBuildIsNoop`. Still not sure why.